### ISSUE TITLE
an example  that einsum_path does not work 

### ISSUE THREAD

**Issue by pingzou** [numpy/numpy#9788]
The function "einsum_path"  does not work for the following example:
```
from numpy import einsum_path
from numpy.random import rand
A1 = rand(9,5,9)
A2= rand(5,5,2,2)
A3 = rand(5,5,2,2)
A4 = rand(9,5,9)
A5 = rand(9,2,2,9)
info1=einsum_path('abc,bdih,dfjm,kfg,chmg->aijk',
                 A1, A2, A3, A4, A5, optimize='optimal')
info2=einsum_path('abc,bdih,dfjm,kfg,chmg->aijk',
                 A1, A2, A3, A4, A5, optimize='greedy')
```
The info1 and info2 don't give any useful information. If changing all "9"  in the above
codes to other integer, the same results will be gotten.
```
print(info1[0])
['einsum_path', (0, 1, 2, 3, 4)]
##
print(info2[0])
['einsum_path', (1, 2), (0, 1, 2, 3)]
```
test on numpy-1.13.1, python-3.62

**Comment by eric-wieser** [comment]
Are you complaining that `optimal` is wrong, `greedy` is wrong, or both are wrong?

**Comment by dgasmith** [comment]
The fact that they do not return the same result is a bug that was recently fixed in #9773. Can verify that this now works under the current HEAD. For the other issue, the intermediates are automatically sieved based off largest array size, giving the path more memory results in:

```
>>> info1 = einsum_path('abc,bdih,dfjm,kfg,chmg->aijk',
                 A1, A2, A3, A4, A5, optimize=('optimal', 1.e6))
>>> print(info1[1])
  Complete contraction:  abc,bdih,dfjm,kfg,chmg->aijk
         Naive scaling:  11
     Optimized scaling:  7
      Naive FLOP count:  1.050e+08
  Optimized FLOP count:  1.231e+05
   Theoretical speedup:  852.625
  Largest intermediate:  1.620e+03 elements
--------------------------------------------------------------------------
scaling                  current                                remaining
--------------------------------------------------------------------------
   6             chmg,abc->hmbag                bdih,dfjm,kfg,hmbag->aijk
   7           hmbag,bdih->imdag                     dfjm,kfg,imdag->aijk
   7           imdag,dfjm->ijfag                          kfg,ijfag->aijk
   6             ijfag,kfg->aijk                               aijk->aijk
```

The greedy algorithm will not find this due to the extremely large outer products required.

**Comment by charris** [comment]
Closing then,
